wgengine/monitor/monitor.go

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package monitor provides facilities for monitoring network
+// interface changes
+package monitor
+
+import (
+	"time"
+
+	"tailscale.com/logger"
+)
+
+// Message represents a message returned from a connection.
+// TODO(]|[): currently messages are being discarded, so the
+// properties of the message haven't been defined.
+type Message interface{}
+
+// Conn represents the connection that is being monitored.
+type Conn interface {
+	Close() error
+	Receive() (Message, error)
+}
+
+// ChangeFunc is a callback function that's called when
+// an interface status changes.
+type ChangeFunc func()
+
+// Mon represents a monitoring instance.
+type Mon struct {
+	logf   logger.Logf
+	cb     ChangeFunc
+	conn   Conn
+	change chan struct{}
+	stop   chan struct{}
```

Comment:
this is a close-only channel, or? where is this closed? meant to close it on close?

Reply:
yeah the stop channel should've been closed on Close()